File: codewars/8kyu/expressions-matter/dduck-expressions-matter.js
// https://www.codewars.com/kata/expressions-matter

const expressionMatter = (a, b, c) => [a * (b + c), a * b * c, a + b * c, (a + b) * c, a + b + c]
  .reduce((acc, cur) => acc < cur ? cur : acc, 0);

// comment
// best practice use Math.max

// function expressionMatter(a, b, c) {
//   return Math.max(
//     a + b + c,
//     a * b * c,
//     a * (b + c),
//     (a + b) * c,
//     a + b * c,
//     a * b + c,
//   );
// }


console.log(expressionMatter(2, 1, 2), 6);
console.log(expressionMatter(2, 1, 1), 4);
console.log(expressionMatter(1, 1, 1), 3);
console.log(expressionMatter(1, 2, 3), 9);
console.log(expressionMatter(1, 3, 1), 5);
console.log(expressionMatter(2, 2, 2), 8);
console.log(expressionMatter(5, 1, 3), 20);
console.log(expressionMatter(3, 5, 7), 105);
console.log(expressionMatter(5, 6, 1), 35);

console.log(expressionMatter(1, 6, 1), 8);

console.log(expressionMatter(2, 6, 1), 14);
console.log(expressionMatter(6, 7, 1), 48);
console.log(expressionMatter(2, 10, 3), 60);
console.log(expressionMatter(1, 8, 3), 27);
console.log(expressionMatter(9, 7, 2), 126);
console.log(expressionMatter(1, 1, 10), 20);
console.log(expressionMatter(9, 1, 1), 18);
console.log(expressionMatter(10, 5, 6), 300);
console.log(expressionMatter(1, 10, 1), 12);

File: codewars/8kyu/expressions-matter/dduck-expressions-matter.py
# https://www.codewars.com/kata/expressions-matter

def expression_matter(a, b, c):
    return max([a * (b + c), a * b * c, a + b * c, (a + b) * c, a + b + c])

print(expression_matter(2, 1, 2), 6)
print(expression_matter(2, 1, 1), 4)
print(expression_matter(1, 1, 1), 3)
print(expression_matter(1, 2, 3), 9)
print(expression_matter(1, 3, 1), 5)
print(expression_matter(2, 2, 2), 8)
print(expression_matter(5, 1, 3), 20)
print(expression_matter(3, 5, 7), 105)
print(expression_matter(5, 6, 1), 35)
print(expression_matter(1, 6, 1), 8)
print(expression_matter(2, 6, 1), 14)
print(expression_matter(6, 7, 1), 48)
print(expression_matter(2, 10, 3), 60)
print(expression_matter(1, 8, 3), 27)
print(expression_matter(9, 7, 2), 126)
print(expression_matter(1, 1, 10), 20)
print(expression_matter(9, 1, 1), 18)
print(expression_matter(10, 5, 6), 300)
print(expression_matter(1, 10, 1), 12)
